app/src/services/cpk.ts

```diff
@@ -1443,16 +1444,23 @@ class CPKService {
     }
   }
 
-  proxyIsUpToDate = async (): Promise<boolean> => {
+  proxyIsUpToDate = async (isNative = false): Promise<boolean> => {
+    if (this.cpk.relay) {
+      return true
+    }
     const network = await this.provider.getNetwork()
     const deployed = await this.cpk.isProxyDeployed()
     if (deployed) {
       const implementation = await this.safe.getMasterCopy()
       if (implementation.toLowerCase() === getTargetSafeImplementation(network.chainId).toLowerCase()) {
         return true
       }
+      return false
     }
-    return false
+    if (isNative) {
+      return false
+    }
+    return true
```

Comment:
Why only false if native?

Reply:
edge case galore 😩 if the proxy has not been deployed but the user wants to interact with a native market we must deploy/upgrade first because both mainnet and rinkeby proxy factories don't support native transactions.

If the first market a user wants to interact with is a standard market then that is fine and we can flag it as upgraded.